exes/mqtt-broadcaster/inc/mqtt_broadcaster.hpp

```diff
@@ -0,0 +1,305 @@
+        mqtt_password_(std::move(mqtt_password)), ctx_(ctx), ipc_client_(ipc_client), mqtt_client_(std::move(mqtt_client)),
+        logger_("mqtt_broadcaster"),
+        dbus_connection_(std::make_unique<sdbusplus::asio::connection>(ctx, tfc::dbus::sd_bus_open_system())),
+        signal_updates_(std::make_unique<sdbusplus::bus::match::match>(
+            *dbus_connection_,
+            sdbusplus::bus::match::rules::propertiesChanged(object_path_, interface_name_),
+            std::bind_front(&mqtt_broadcaster::update_signals, this))),
+        config_(ctx, "mqtt_broadcaster") {
+    asio::co_spawn(mqtt_client_->strand(), initialize(), asio::detached);
+  }
+
+private:
+  auto initialize() -> asio::awaitable<void> {
+    // Initial connect to the broker
+    if (!connect_active_) {
+      connect_active_ = true;
+      co_await asio::co_spawn(mqtt_client_->strand(), connect_to_broker(), asio::use_awaitable);
+    }
+
+    // When a new list of banned string arrives the program is restarted
+    config_.value()._banned_topics.observe([this](auto& new_conf, [[maybe_unused]] auto& old_conf) {
+      banned_signals_.clear();
+      for (auto const& con : new_conf) {
+        banned_signals_.push_back(con);
+      }
+      restart();
+    });
+
+    asio::co_spawn(mqtt_client_->strand(), tfc::base::exit_signals(ctx_), asio::detached);
+
+    banned_signals_ = config_.value()._banned_topics.value();
+    load_signals();
+  }
+
+  // Connect the mqtt client to the broker. Only a single "thread" can run this at a time. The routine that calls this
+  // function must set the connect_active_ flag to true before calling this function and the function sets it to false when
+  // finished.
+  auto connect_to_broker() -> asio::awaitable<void> {
+    while (true) {
+      try {
+        logger_.trace("Connecting to MQTT broker");
+        asio::ip::tcp::socket resolve_sock{ ctx_ };
+        asio::ip::tcp::resolver res{ resolve_sock.get_executor() };
+
+        logger_.trace("Resolving IP number");
+        asio::ip::tcp::resolver::results_type resolved_ip =
+            co_await res.async_resolve(mqtt_host_, mqtt_port_, asio::use_awaitable);
+
+        logger_.trace("Connecting MQTT client socket to IP number of broker");
+        co_await asio::async_connect(mqtt_client_->next_layer(), resolved_ip, asio::use_awaitable);
+
+        logger_.trace("Sending Connect request to the MQTT broker");
+
+        // Connect is initialized with clean start set to false, which means that the broker will keep the session alive even
+        // if the client disconnect, and it will keep that session alive for UINT_MAX seconds.
+        auto connect_packet =
+            async_mqtt::v5::connect_packet{ false,
+                                            std::chrono::seconds(100).count(),
+                                            async_mqtt::allocate_buffer("cid1"),
```

Comment:
could we make a better client id than cid1? What happens if two PCs connect with the same cliend id to one mqtt broker?

Reply:
How about a MAC address or a randomly generated hash?

Reply:
how about the hostname?

Reply:
local ip number or public ip? local would probaby be more beneficial since these hosts are most often running on the same LAN.